### ISSUE TITLE
Create a host symbol for host RID (or RID components) for use in templates

### ISSUE THREAD

**Issue by baronfel** [dotnet/templating#8307]
### Is your feature request related to a problem? Please describe.

Today we provide the following Host-provided symbols for binding:

* HostIdentifier
* WorkingDirectory

We should make it easier for template authors to create per-architecture defaults/options by providing the current host RID as another symbol/bindable value.

### Describe the solution you'd like.

We should provide two new symbols to allow template authors to condition code/assets/symbols/etc based on host OS and host architecture. 

* HostPlatform // e.g. win, macos, linux - the first part of the RID. May be dynamic because it is powered by the RID graph.
* HostArchitecture // eg. x64, arm64, x85 - the second logical part of the RID. May be dynamic because it is powered by the RID graph.

**Comment by marcpopMSFT** [comment]
I assume you are proposing Host* for these to be consistent with the existing Identifier one?

**Comment by baronfel** [comment]
Yeah, exactly - then if we needed an override or to change/let the user specify a different one there might be a Target* variant of those same properties.

**Comment by jsturtevant** [comment]
I would find this useful as well.  I tried using the `OS` as described in https://github.com/dotnet/templating/discussions/7405 but found that it only works on Windows. In my use case I need to know if I am on `Linux` or `MacOs` as well.
